pkg/api/getambassador.io/v2/common.go

```diff
@@ -143,6 +145,51 @@ type ErrorResponseOverride struct {
 	Body ErrorResponseOverrideBody `json:"body,omitempty"`
 }
 
+type StatusRange struct {
+	// Start of the statuses to include. Must be between 100 and 599 (inclusive)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=100
+	// +kubebuilder:validation:Maximum=599
+	Start int `json:"start,omitempty"`
+	// End of the statuses to include. Must be between 100 and 599 (inclusive)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=100
+	// +kubebuilder:validation:Maximum=599
+	End int `json:"end,omitempty"`
+}
+
+type HealthCheck struct {
+	// Timeout for connecting to the health checking endpoint. Defaults to 3 seconds.
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	// Interval between health checks. Defaults to every 5 seconds.
+	Interval *metav1.Duration `json:"interval,omitempty"`
+	// Number of non-expected responses for the upstream to be considered unhealthy. A single 503 will mark the upstream as unhealthy. Defaults to 2.
+	UnhealthyThreshold *int `json:"unhealthy_threshold,omitempty"`
+	// Number of expected responses for the upstream to be considered healthy. Defaults to 1.
+	HealthyThreshold *int             `json:"healthy_threshold,omitempty"`
```

Comment:
Is there a reason we are using underscores and not camel casing? I know I'm being slow on nailing down CRD design guidelines but since this is a new one I would think we would change it to `healthyThreshold`...For example, here is the K8s guidelines and when we get to V3 we want to align with it.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions

Reply:
This was actually something I was hoping to sync about regarding the next major CRD version, particularly whether it would be better for this whole health_checks object to follow the syntax of the other Mapping fields for now, then change everything all at once with the next CRD version, or if there is a benefit to mixing the casing until everything is camelCase.

Reply:
Yes, I think that is a good call for this one and agree. So the rule for now will be:

If adding to an existing CRD as in this case extending the `Mapping` then we should match existing conventions. If it is a new standalone CRD then we should adopt k8s naming guidelines. Then in the next version and ultimately when we get to v3 then all the crds will be aligned on the new naming conventions.

I will document this in the design guide for v3. I swear I'm going to finish it...someday :).

Reply:
Cool. I'm definitely don't have any strong opinions on this, just want to make sure there is a standard to follow and that we go with whatever eliminates the most friction.